drivers/media/i2c/imx290.c

```diff
@@ -1115,6 +1137,17 @@ static int imx290_probe(struct i2c_client *client)
 		return -ENODEV;
 	}
 
+	match = of_match_device(imx290_of_match, dev);
+	if (!match)
+		return -ENODEV;
+	if (!match->data) {
```

Comment:
It's shame you can't just use the match data directly, but those ARRAY_SIZE()s make it hard to find a neat way to do it.

Reply:
There are only ever two formats, so the arrays should always be the same size.

If I can remember the compile time array size check macros to enforce that then I'd happily switch to passing the format pointer directly from match data.

Reply:
Actually
```
#define IMX290_NUM_FORMATS 2

static const struct imx290_pixfmt imx290_colour_formats[IMX290_NUM_FORMATS] = {
	{ MEDIA_BUS_FMT_SRGGB10_1X10, 10 },
	{ MEDIA_BUS_FMT_SRGGB12_1X12, 12 },
};

static const struct imx290_pixfmt imx290_mono_formats[IMX290_NUM_FORMATS] = {
	{ MEDIA_BUS_FMT_Y10_1X10, 10 },
	{ MEDIA_BUS_FMT_Y12_1X12, 12 },
};
```
and then using IMX290_NUM_FORMATS instead of the equivalent ARRAY_SIZE value. If someone were to ever add an extra value into the array then it'd break the compile.

Happy with that solution?

Reply:
Delighted.

Reply:
Done then.